presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -229,20 +231,20 @@ private static boolean isPrestoView(Table table)
     }
 
     @Override
-    public Optional<Set<ColumnStatisticsObj>> getTableColumnStatistics(String databaseName, String tableName, Set<String> columnNames)
+    public Set<ColumnStatisticsObj> getTableColumnStatistics(String databaseName, String tableName, Set<String> columnNames)
     {
         try {
             return retry()
                     .stopOn(NoSuchObjectException.class, HiveViewNotSupportedException.class)
                     .stopOnIllegalExceptions()
                     .run("getTableColumnStatistics", stats.getGetTableColumnStatistics().wrap(() -> {
                         try (HiveMetastoreClient client = clientProvider.createMetastoreClient()) {
-                            return Optional.of(ImmutableSet.copyOf(client.getTableColumnStatistics(databaseName, tableName, ImmutableList.copyOf(columnNames))));
+                            return ImmutableSet.copyOf(client.getTableColumnStatistics(databaseName, tableName, ImmutableList.copyOf(columnNames)));
                         }
                     }));
         }
         catch (NoSuchObjectException e) {
-            return Optional.empty();
+            return ImmutableSet.of();
```

Comment:
This again shows that we no longer distinguish "normal path" and "bad state" in a clean way.

Reply:
How is that different? Both `Optional.empty()` and `ImmutableSet.of()` mean that there's no statistics? It seems that the right thing to do here is to throw an exception. I just decided to preserve the behaviour, introduced obviously for a reason.

Reply:
As I understand `Optional.empty()` means we do not know the stats, while empty set means that stats are known and they are empty.

Reply:
From what i understand the `Optional.empty()` means `we don't know the stats`. And the `ImmutableSet.of()` means that `we know stats for 0 columns`. If we deliberately want to put `null` for a particular stat, say for the `min` or the `max` if the table is empty, we would still have a non empty with a `ColumnStatisticsObj` object that has it's `min` and `max` fields set to null.

Reply:
@kokosing  I think the question to consider is whether the behavior should ever be different between the two cases.  Previously the resulting behavior was exactly the same but it was harder to tell because there was an extra layer of misdirection.  It also doesn't seem like we'll want to distinguish between the two anytime in the near future (for our purposes the statistics aren't known, who cares whether there was an empty object in the metastore), so simplifying things seems like a good idea to me.